File: client/neat-client-python/neat_client_python/model/token.py
from typing import NamedTuple, Optional
from http import HTTPStatus
from os.path import join

from . import BaseModel, HTTPMethod, API_PREFIX


class GetTokenRequest(NamedTuple):
    username: str
    password: str


class TokenRespose(NamedTuple):
    token: str


class Token(BaseModel):
    uri = join(API_PREFIX, "token/")
    auth = False
    method = HTTPMethod.POST.value
    status_code = HTTPStatus.OK
    request_class = GetTokenRequest
    response_class = TokenRespose

    def __init__(self, request_data):
        self.request_data = request_data


class VerifyTokenRequest(NamedTuple):
    token: str


class VerifyToken(BaseModel):
    uri = join(API_PREFIX, "token/verify")
    auth = False
    method = HTTPMethod.POST.value
    status_code = HTTPStatus.OK
    request_class = VerifyTokenRequest
    response_class = TokenRespose

    def __init__(self, request_data):
        self.request_data = request_data

File: client/neat-client-go/token.go
package client

import (
	"context"
	"io"
	"net/http"
)

var (
	TOKEN_URI = JoinURL(API_PREFIX, "/token/")
)

type GetTokenRequest struct {
	BaseRequest
	Username string `json:"username"`
	Password string `json:"password"`
}

func (r *GetTokenRequest) ToPostData() (io.Reader, error) {
	return r.toPostData(r)
}

type TokenResponse struct {
	BaseResponse
	Token string `json:"token"`
}

func (r *GetTokenRequest) GetRequestData() *RequestDate {
	if r.RequestDate == nil {
		r.RequestDate = &RequestDate{
			Auth:       false,
			Method:     http.MethodPost,
			URI:        TOKEN_URI,
			StatusCode: http.StatusOK,
		}
	}
	return r.RequestDate
}

func (r *TokenResponse) ToStruct(reader io.Reader) error {
	return r.toStruct(r, reader)
}

func (c *NeatClient) GetToken(ctx context.Context, header map[string]string, request *GetTokenRequest) (*TokenResponse, error) {
	response := &TokenResponse{}
	err := c.send(ctx, header, request, response)
	return response, err
}

type VerifyTokenRequest struct {
	BaseRequest
	Token string `json:"token"`
}

func (r *VerifyTokenRequest) GetRequestData() *RequestDate {
	if r.RequestDate == nil {
		r.RequestDate = &RequestDate{
			Auth:       false,
			Method:     http.MethodPost,
			URI:        JoinURL(TOKEN_URI, "/verify"),
			StatusCode: http.StatusOK,
		}
	}
	return r.RequestDate
}

func (r *VerifyTokenRequest) ToPostData() (io.Reader, error) {
	return r.toPostData(r)
}

func (c *NeatClient) VerifyToken(ctx context.Context, header map[string]string, request *VerifyTokenRequest) (*TokenResponse, error) {
	response := &TokenResponse{}
	err := c.send(ctx, header, request, response)
	return response, err
}
